samtranslator/model/sam_resources.py

```diff
@@ -2407,3 +2420,197 @@ def _set_resolver_code_settings(resolver_code_settings: aws_serverless_graphqlap
+        """
+        if function.DataSourceName:
+            return function.DataSourceName
+
+        if function.DataSource in NONE_DATASOURCE_LITERALS:
+            return none_datasource_logical_id
+
+        if not datasources or function.DataSource not in datasources:
+            raise InvalidResourceException(
+                relative_id, f"No DynamoDB DataSource exists with logical id '{function.DataSource}'."
+            )
+
+        return datasources[function.DataSource].Name or function.DataSource
+
+    @staticmethod
+    def _parse_function_sync(sync: aws_serverless_graphqlapi.Sync) -> SyncConfigType:
+        sync_config: SyncConfigType = {"ConflictDetection": passthrough_value(sync.ConflictDetection)}
+
+        if sync.ConflictHandler:
+            sync_config["ConflictHandler"] = passthrough_value(sync.ConflictHandler)
+
+        if sync.LambdaConflictHandlerArn:
+            sync_config["LambdaConflictHandlerConfig"] = {
+                "LambdaConflictHandlerArn": passthrough_value(sync.LambdaConflictHandlerArn)
+            }
+
+        return sync_config
+
+    @staticmethod
+    def _parse_code_properties(
+        resource: aws_serverless_graphqlapi.Function,
+        code_settings: Optional[aws_serverless_graphqlapi.ResolverCodeSettings],
+        name: str,
+        relative_id: str,
+    ) -> Tuple[Optional[str], Optional[str]]:
+        """
+        Parses the code properties from Serverless::GraphQLApi function (ADD RESOLVER AFTER).
+
+        This function parses the "CodeUri" and "InlineCode" properties for Function (AND RESOLVER) resources.
+        It also raises exceptions when the customer template is invalid. The return is a tuple of the values
+        (InlineCode, CodeUri).
+        """
+        if resource.InlineCode and resource.CodeUri:
+            raise InvalidResourceException(
+                relative_id, "Both 'InlineCode' and 'CodeUri' cannot be defined at the same time."
+            )
+
+        if resource.InlineCode:
+            return resource.InlineCode, None
+
+        if resource.CodeUri:
+            return None, resource.CodeUri
+
+        if not code_settings:
+            raise InvalidResourceException(
+                relative_id, "One of 'InlineCode', 'CodeUri', or 'ResolverCodeSettings' must be set."
+            )
+
+        # TODO: replace "name" with "FileNamePattern" from resolver_code_settings once implemented
+        # TODO: return the path with ResolversFolder if type is Resolver once we implement that
+        return None, f"{code_settings.CodeRootPath}/{code_settings.FunctionsFolder}/{name}.js"
+
+    @staticmethod
+    def _parse_runtime(
+        resource: aws_serverless_graphqlapi.Function,
+        code_settings: Optional[aws_serverless_graphqlapi.ResolverCodeSettings],
+        relative_id: str,
+    ) -> AppSyncRuntimeType:
+        # Runtime can exist in either the resource or in ResolverCodeSettings as a default.
```

Comment:
Why does `Function` take precedence?

(tangential nit: `resource` is quite overloaded, here it's the `Function` property specifically so I'd rather call it that, similar to `code_settings`)

Reply:
I plan on using this method for Resolver as well once implemented since in context of this function it would be the exact same.

Reply:
Resolver also has the `Runtime` property. Let me add a 1 line comment

Reply:
It's still not clear to me why `resource.Runtime` takes precedence over `code_settings.Runtime`. Is it "just because", or is there some other rationale?

Reply:
Ah, I misinterpreted what you were asking. ResolverCodeSettings is where the customer will set defaults that will apply to a majority of the Functions/Resolvers they define. In the case where they have a resource that does not want to use the ResolverCodeSettings they can overwrite by defining it themselves.

Reply:
In `resource.Runtime` customers can override their `code_settings` for API for a single function. For example, once AppSync starts supporting typescript, `code_settings` can be `TS` but then customers can make a single Function in `JS`